stratosphere/ams_mitm/source/fs_mitm/fsmitm_service.cpp

```diff
@@ -261,11 +262,21 @@ Result FsMitmService::OpenBisStorage(Out<std::shared_ptr<IStorageInterface>> out
             const bool is_sysmodule = TitleIdIsSystem(this->title_id);
             const bool has_bis_write_flag = Utils::HasFlag(this->title_id, "bis_write");
             const bool has_cal0_read_flag = Utils::HasFlag(this->title_id, "cal_read");
+            const bool has_blank_cal0_flag = Utils::HasGlobalFlag("blank_prodinfo");
             if (bis_partition_id == BisStorageId_Boot0) {
                 storage = std::make_shared<IStorageInterface>(new Boot0Storage(bis_storage, this->title_id));
             } else if (bis_partition_id == BisStorageId_Prodinfo) {
-                /* PRODINFO should *never* be writable. */
-                if (is_sysmodule || has_cal0_read_flag) {
+                if (has_blank_cal0_flag) {
+                    FsFile file;
+
+                    rc = Utils::OpenBlankProdinfoFile(&file);
+                    if (R_FAILED(rc)) {
+                        return rc;
+                    }
+                    
+                    storage = std::make_shared<IStorageInterface>(new FileStorage(new ProxyFile(&file))); /* Should we make this read-only? */
```

Comment:
It's already read-only because it's opened FS_OPEN_READ.

Also, this still leaks object IDs, but I'll just fix that when it's merged as explaining why might be complicated.